File: core/core-persistence-jooq/src/main/java/ch/difty/scipamato/core/persistence/newsletter/NewsletterTopicFilterConditionMapper.java
package ch.difty.scipamato.core.persistence.newsletter;

import java.util.List;

import org.jetbrains.annotations.NotNull;
import org.jooq.Condition;

import ch.difty.scipamato.common.persistence.AbstractFilterConditionMapper;
import ch.difty.scipamato.common.persistence.FilterConditionMapper;
import ch.difty.scipamato.core.entity.newsletter.NewsletterTopicFilter;

/**
 * Mapper turning the provider {@link NewsletterTopicFilter} into a jOOQ
 * {@link Condition}.
 *
 * @author u.joss
 */
@FilterConditionMapper
public class NewsletterTopicFilterConditionMapper extends AbstractFilterConditionMapper<NewsletterTopicFilter> {

    @Override
    public void map(@NotNull final NewsletterTopicFilter filter, @NotNull final List<Condition> conditions) {
        // currently no fields to be handled
    }
}

File: core/core-persistence-jooq/src/main/java/ch/difty/scipamato/core/persistence/newsletter/NewsletterTopicFilterConditionMapper.kt
package ch.difty.scipamato.core.persistence.newsletter

import ch.difty.scipamato.common.persistence.AbstractFilterConditionMapper
import ch.difty.scipamato.common.persistence.FilterConditionMapper
import ch.difty.scipamato.core.entity.newsletter.NewsletterTopicFilter
import org.jooq.Condition

/**
 * Mapper turning the provider [NewsletterTopicFilter] into a jOOQ [Condition].
 */
@FilterConditionMapper
class NewsletterTopicFilterConditionMapper : AbstractFilterConditionMapper<NewsletterTopicFilter>() {

    // currently no fields to be handled
    override fun internalMap(filter: NewsletterTopicFilter): List<Condition> = emptyList()
}
